File: sample-rest-api/src/test/java/nl/haploid/octowight/sample/AbstractIT.java
package nl.haploid.octowight.sample;

import nl.haploid.octowight.sample.configuration.PostgresConfiguration;
import nl.haploid.octowight.sample.configuration.TestConfiguration;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
import org.springframework.test.context.support.AnnotationConfigContextLoader;

@ContextConfiguration(
		classes = {TestConfiguration.class, PostgresConfiguration.class},
		loader = AnnotationConfigContextLoader.class)
public abstract class AbstractIT extends AbstractTransactionalJUnit4SpringContextTests {
}

File: sample-rest-api/src/test/scala/nl/haploid/octowight/sample/AbstractIT.scala
package nl.haploid.octowight.sample

import nl.haploid.octowight.sample.configuration.{PostgresConfiguration, TestConfiguration, WebMvcConfiguration}
import org.scalatest._
import org.scalatest.tags.Slow
import org.springframework.test.context.web.{AnnotationConfigWebContextLoader, WebAppConfiguration}
import org.springframework.test.context.{ContextConfiguration, TestContextManager}

@ContextConfiguration(classes = Array(classOf[TestConfiguration], classOf[PostgresConfiguration], classOf[WebMvcConfiguration]),
  loader = classOf[AnnotationConfigWebContextLoader])
@WebAppConfiguration
@Slow
abstract class AbstractIT extends FlatSpec with BeforeAndAfterEach with ShouldMatchers {

  override def beforeEach() = new TestContextManager(this.getClass).prepareTestInstance(this)
}
